src/ops/fuelup_toolchain/export.rs

```diff
@@ -0,0 +1,331 @@
+                }
+            }
+        }
+    }
+    let mut export_channel = channel.unwrap_or(toolchain_name.clone());
+    if toolchain_override::Channel::from_str(&export_channel).is_err() {
+        println_warning(&format!(
+            "Invalid channel '{}', expected one of {}. \
+            Please input a valid channel: ",
+            export_channel, VALID_CHANNEL_STR,
+        ));
+        let mut input_channel_name = String::new();
+        if reader.read_line(&mut input_channel_name).is_err() {
+            bail!("failed to read user input");
+        }
+        input_channel_name = String::from(input_channel_name.trim());
+        if toolchain_override::Channel::from_str(&input_channel_name).is_err() {
+            bail!(
+                "Invalid channel '{}', expected one of {}.",
+                input_channel_name,
+                VALID_CHANNEL_STR,
+            );
+        } else {
+            export_channel = input_channel_name;
+        }
+    }
+
+    let toolchain_override = ToolchainOverride {
+        cfg: OverrideCfg::new(
+            ToolchainCfg {
+                // here shouldn't be err as we has checked above
+                channel: toolchain_override::Channel::from_str(&export_channel)?,
+            },
+            Some(version_map),
+        ),
+        path: toolchain_info_path.clone(),
+    };
+    let document = toolchain_override.to_toml();
+    if std::fs::write(toolchain_override.path, document.to_string()).is_err() {
+        bail!("failed to write {}", toolchain_info_path.display());
+    }
+
+    info!(
+        "exported '{}' into '{}'",
+        toolchain_name,
+        toolchain_info_path.display()
+    );
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        channel, path::settings_file, toolchain::DistToolchainName,
+        toolchain_override::ToolchainOverride,
+    };
+    use serial_test::serial;
+    use std::{fs, io::Write};
+
+    use super::*;
+
+    // simulate input
+    const INPUT_NOP: &[u8; 1] = b"\n";
+    const INPUT_YES: &[u8; 2] = b"y\n";
+    const INPUT_NO: &[u8; 2] = b"n\n";
+    const INPUT_INVALID_CHANNEL: &[u8; 11] = b"my-channel\n";
+    const INVALID_CHANNEL: &str = "my-channel";
+
+    fn remove_toolchain_info() {
+        if let Some(fuel_toolchain_toml_file) = get_fuel_toolchain_toml() {
+            if fuel_toolchain_toml_file.exists() {
+                fs::remove_file(fuel_toolchain_toml_file).unwrap();
+            }
+        }
+    }
+    fn create_toolchain_info() {
+        if let Some(fuel_toolchain_toml_file) = get_fuel_toolchain_toml() {
+            if !fuel_toolchain_toml_file.exists() {
+                fs::create_dir_all(fuel_toolchain_toml_file.parent().unwrap()).unwrap();
+                fs::File::create(&fuel_toolchain_toml_file).unwrap();
+            }
+        } else {
+            fs::File::create(FUEL_TOOLCHAIN_TOML_FILE).unwrap();
+        }
+    }
+    // mock setting.toml
+    fn create_toolchain_settings_file() {
```

Comment:
I believe these helpers touch the disk i.e. create and write a file and then remove it.
Does Rust have a library for doing this all in-memory instead? If it does then we could pass the handle around and inspect it instead of reading / writing when testing and potentially leaving behind some files if something exits early.

This is coming from the perspective of having done something like this in Python before

Reply:
I'm not sure. But `Export`function inner will read file from real disk not the memory.

Reply:
I think it's okay for the tests to read/write to disk, but the tests should clean up after themselves.